core/common/src/main/java/alluxio/grpc/GrpcConnectionPool.java

```diff
@@ -276,40 +276,41 @@ private void shutdownManagedChannel(ManagedChannel managedChannel, AlluxioConfig
 
   private EventLoopGroup acquireNetworkEventLoop(GrpcChannelKey channelKey,
       AlluxioConfiguration conf) {
-    return mEventLoops.compute(channelKey, (k, v) -> {
+    return mEventLoops.compute(channelKey.getNetworkGroup(), (key, v) -> {
       // Increment and return if event-loop found.
       if (v != null) {
-        LOG.debug("Acquiring an existing event-loop. ChannelKey: {}. Ref-Count:{}", channelKey,
-            v.getRefCount());
+        LOG.debug("Acquiring an existing event-loop for {}. Ref-Count:{}",
+            channelKey, v.getRefCount());
         v.reference();
         return v;
       }
 
       // Create a new event-loop.
       ChannelType nettyChannelType =
           NettyUtils.getChannelType(PropertyKey.Template.USER_NETWORK_NETTY_CHANNEL
-              .format(channelKey.getNetworkGroup().getPropertyCode()), conf);
+              .format(key.getPropertyCode()), conf);
       int nettyWorkerThreadCount =
           conf.getInt(PropertyKey.Template.USER_NETWORK_NETTY_WORKER_THREADS
-              .format(channelKey.getNetworkGroup().getPropertyCode()));
```

Comment:
Let's say we have 2 clients with different config for the netty worker threads. If one client is already active, the 2nd client's thread config will be ignored, correct?

Reply:
Yes. Connection level properties should be consistent.

Reply:
That behavior seems a little sketchy to me

Reply:
How about we bring singleton back for a subset of props?

Reply:
I don't really see a good way to do this unless we make the # of threads unconfigurable/unbounded. Otherwise if we expose the configuration to users it will be kind of confusing. Either way, I think that is fine to work on in a later PR.

Reply:
How about `CheckConsistency.ENFORCE` ?

Reply:
Consistency level won't prevent an application in a client JVM writing `FileSystem.Factory.create(conf)` where the conf has two separate values for the netty threadpool